sweepbatcher/presigned.go

```diff
@@ -389,21 +392,31 @@ func presign(ctx context.Context, presigner presigner, destAddr btcutil.Address,
 
 		// Try to presign this transaction.
 		const loadOnly = false
-		_, err = presigner.SignTx(
-			ctx, primarySweepID, tx, batchAmt, minRelayFeeRate, fr,
-			loadOnly,
-		)
-		if err != nil {
-			return fmt.Errorf("failed to presign unsigned tx %v "+
-				"for feeRate %v: %w", tx.TxHash(), fr, err)
-		}
+		eg.Go(func() error {
+			_, err := presigner.SignTx(
+				grCtx, primarySweepID, tx, batchAmt,
+				minRelayFeeRate, fr, loadOnly,
+			)
+			if err != nil {
+				return fmt.Errorf("failed to presign unsigned "+
```

Comment:
before we return, should we cancel `grCtx` so the other signing requests are stopped?

Reply:
@GeorgeTsagk Each goroutine signs a separate transaction (they are the same except that they have different fee rates). We need all of them to proceed. If some of them fail to sign, we cancel the whole process.